### PR TITLE
Osd 20178 routine upgrade

### PR DESCRIPTION
**Pull request by tkong-redhat**
### What type of PR is this?
Quarterly upgrade,

- Upgrade go to 1.20 to support OCP 4.14
- Upgrade controller runtime
- Doc update

### What this PR does / why we need it?
Quarterly maintenance work for ocm-agent-operator

### Which Jira/Github issue(s) this PR fixes?
[OSD-20178](https://issues.redhat.com//browse/OSD-20178)

### Pre-checks (if applicable):
- [x] Tested latest changes against a cluster
- [ ] Ran `make generate` command locally to validate code changes
- [ ] Included documentation changes with PR

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,97 +1,101 @@
...
-	k8s.io/api v0.27.4
-	k8s.io/apimachinery v0.27.4
-	k8s.io/client-go v0.27.4
-	k8s.io/kube-openapi v0.0.0-20230501164219-8b0f38b5fd1f
-	sigs.k8s.io/controller-runtime v0.15.0
-	sigs.k8s.io/controller-tools v0.11.3
+	go.uber.org/zap v1.26.0
+	k8s.io/api v0.29.1
+	k8s.io/apimachinery v0.29.1
+	k8s.io/client-go v0.29.1
```
Comment (ravitri) [inline go.mod:19]: Can we stick to version 0.27.x (x being latest available in these deps) for api, apimachinery and client-go for Kube? That's the Kube version for OCP 4.14 which we want to stick with for now..

Comment (ravitri) [reply go.mod:19]: Reference - https://docs.openshift.com/container-platform/4.14/updating/preparing_for_updates/updating-cluster-prepare.html#kube-api-removals_updating-cluster-prepare

Comment (tkong-redhat) [reply go.mod:19]: Sure, good spot. I'll make the update.

**Comment by tkong-redhat** [comment]
Test passed on OCP 4.14.12 and 4.15.0-rc7

**Pull request merged**